### PR TITLE
Added normalized_epsilon as a parameter to convolve_fft (replaces #5177)

### PR DESCRIPTION
**Pull request by mwcraig**
A replacement for #5177 since a force push to that branch failed.

@plim -- I have not done any code review on this, simply rebased and addressed the comments of @astrofrog and @MSeifert04 in #5177.

### CODE REVIEW

**Comment by mwcraig** [comment]
The failure on travis are with numpy 1.12, see: https://github.com/astropy/astropy/issues/5629

**Comment by pllim** [comment]
@keflavich , @adonath , @larrybradley ?

**Comment by mwcraig** [comment]
Working on adding a test (and will add a changelog entry).

File: astropy/convolution/convolve.py [hunk 1]
```diff
@@ -254,8 +254,8 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
 def convolve_fft(array, kernel, boundary='fill', fill_value=0, crop=True,
                  return_fft=False, fft_pad=None, psf_pad=None,
                  interpolate_nan=False, quiet=False, ignore_edge_zeros=False,
-                 min_wt=0.0, normalize_kernel=False, allow_huge=False,
-                 fftn=np.fft.fftn, ifftn=np.fft.ifftn,
+                 min_wt=0.0, normalize_kernel=False, normalization_rtol=1e-8,
```
Comment (MSeifert04) [inline astropy/convolution/convolve.py]: I know it's just a nitpick but this changes the function signature if the additional parameter isn't added at the end of the list of parameters.

The `mask` was also appended so maybe it wouldn't hurt too much if you place it at the end?

**Comment by keflavich** [comment]
I'm happy with this change.  I'd like to incorporate it into #5782.  Does anyone have a preference on merging it directly here vs merging it into my PR on #5782 and eventually into master?

**Comment by keflavich** [comment]
I can merge this into #5782 but I haven't done so yet;  I have a PR open onto this branch that rebases it.  @mwcraig, thoughts?

**Comment by bsipocz** [comment]
@keflavich - Isn't #5782 almost ready to go, but just waits for Erik to sign off on it? If that's the case I think it's better to leave this here and rebase asap the other is going in.

**Comment by keflavich** [comment]
Yep.  And, that sounds good to me.  I'll help get this PR into shape once 5782 is done.  I know what needs doing.

**Comment by mwcraig** [comment]
Let me know if you need any help @keflavich

**Comment by bsipocz** [comment]
@mwcraig - Could you please rebase? #5782 is merged now.

**Comment by mwcraig** [comment]
@bsipocz -- starting it now...

**Comment by mwcraig** [comment]
Trying again....

**Comment by mwcraig** [comment]
Testing locally before I push again...

**Comment by mwcraig** [comment]
@bsipocz @keflavich -- I'm a little confused. The change in this PR, adding a tolerance for whether a kernel is already normalize when `normalize_kernel` is `False`, comes at https://github.com/astropy/astropy/blob/master/astropy/convolution/convolve.py#L565, reached if `normalize_kernel=False`. 

In that else clause is another `if` that checks whether the kernel is already normalized. If it is, the kernel_scale is set to 1 and the kernel isn't normalized, which makes sense. 

If that test fails, and the kernel does not sum to nearly zero, then the kernel is normalized (https://github.com/astropy/astropy/blob/master/astropy/convolution/convolve.py#L579) even though `normalize_kernel=False`. It seems like that shouldn't happen....

**Comment by mwcraig** [comment]
I gather that if `nan_treatment` is `interpolate` then the kernel will *always* be renormalized, with an error raised if the kernel sums to zero. 

In that case should the combination of `nan_treatment='interpolate'` and `normalize_kernel=False` raise an error right away?

**Comment by mwcraig** [comment]
Cancelling all of the tests except CircleCI because I know the tests will fail. Letting circle complete so you can see the issue...

**Comment by keflavich** [comment]
@mwcraig - yes, the kernel is normalized even if `normalize_kernel` is set to `False`, but the normalization is undone later.  The idea is, someone could request an un-normalized convolution  but still want NaNs interpolated over.  This is possible, as long as the kernel is normalizable.

I added to this PR by including `normalization_rtol` in `convolve` as well, where it's being used in a similar way.  I also added some clarificatory comments.

**Comment by mwcraig** [comment]
@keflavich -- I believe the failures before my most recent push were real. See https://travis-ci.org/astropy/astropy/jobs/244124747#L2543; the problem is that even though `normalize_kernel=False` the kernel ends up being normalized. This happened only if the kernel summed to 1 within with `normalization_rtol`. Then `kernel_scale` is set to 1, but when `nan_handling='interpolate'` the image weighting ended up scaling by the unnormalized kernel and the kernel ends up normalized. See [this line](https://github.com/astropy/astropy/pull/5649/files#diff-f632180d9f1e3fcf689f62e4c424e94fR702), which seems to imply the kernel is assumed to be normalized.

I've rewritten a bit of `convolve_fft` to eliminate the business about skipping the normalization if it is already close enough to 1. With your refactor it seemed unnecessary; regardless of how close the sum of the kernel is to 1, if `normalize_kernel=False` the right thing to do given your changes is to temporarily scale to 1 and then scale back. The tests behave as expected with this change.

I ended up changing the name of the parameter to `normalization_zero_tol` since that is the one place it is left -- in the check of whether the kernel is normalizable. 

If this looks good to you, feel free to merge...

**Comment by mwcraig** [comment]
(assuming tests pass, of course)

**Comment by keflavich** [comment]
The changes look fine to me.  I've added a tiny change for efficiency's sake.  I'll see if I can get coverage up...

**Comment by bsipocz** [comment]
Ignore the coverage, if a PR is not on top of master it will report this decrease. After the release I'll shop around and see whether other coverage services work better than coveralls (I have a few candidates that looked promising, but they need a proper try-out).

**Comment by bsipocz** [comment]
@keflavich @mwcraig - CI passed, can this be merged?

**Comment by mwcraig** [comment]
yes

**Pull request merged**